jwst/fits_generator/create_dms_data.py

```diff
@@ -282,27 +288,33 @@ def flip_rotate(input_hdulist):
         #
         rcube = cube[:, ::-1]
         #
-        # ROWSTART and COLSTART are zero-indexed
+        # ROWSTART and COLSTART are zero-indexed, COLCORNR and ROWCORNR
+        # are 1-indexed
         #
+        detector_row_start = None
         try:
-            detector_row_start = header['ROWSTART']
+            detector_row_start = int(float(header['ROWSTART'])) + 1
         except KeyError:
-            print('Unable to get subarray ROWSTART, using 1')
-            detector_row_start = '0.000000'
+            pass
         try:
-            detector_row_start = int(float(detector_row_start)) + 1
-        except ValueError:
-            print('Unable to convert ROWSTART to a valid integer, using 1')
+            detector_row_start = int(header['ROWCORNR'])
```

Comment:
Can you put a comment as to why `ROWSTART` and `ROWCORNR` are read in as if only one or the other could be present.

Lines 241-272 and 290-321 are the same. Can you combine them, i.e. read in the keywords in one place and put the different code only in if ... else.... The advantage i syou can put the above comment in one place only :)